Source/WTF/wtf/TypeCasts.h

```diff
@@ -127,13 +121,7 @@ inline match_constness_t<Source, Target>* downcast(Source* source)
 {
     static_assert(!std::is_same_v<Source, Target>, "Unnecessary cast to same type");
     static_assert(std::is_base_of_v<Source, Target>, "Should be a downcast");
-    // FIXME: This is too expensive to enable on x86 for now but we should try and
-    // enable the RELEASE_ASSERT() on all architectures.
-#if CPU(ARM64)
     RELEASE_ASSERT(!source || is<Target>(*source));
```

Comment:
Can’t this be written as follows?
```suggestion
    RELEASE_ASSERT(is<Target>(source));
```

Reply:
That doesn't allow source to be null. The status quo does.

Reply:
Yes, I'll update. Thanks.

Reply:
Hmm, actually this wouldn't work.
`RELEASE_ASSERT(is<Target>(source));` is equivalent to `RELEASE_ASSERT(source && is<Target>(*source));`, not `RELEASE_ASSERT(!source || is<Target>(*source));`

Reply:
Oh good. You had me questioning my understanding of things for a moment there.

Reply:
> That doesn't allow source to be null. The status quo does.

Well spotted. I’m glad I worded my suggestion as a question. Less embarrassing that I got it wrong.